# Lab 1.1 — Exploring the Federal AI Use Case Inventory

*Course 1258 — Applied AI for Government IT Professionals · Ch01 AI and ML
Foundations for Government · 30 minutes · CloudShare VM (JupyterLab) ·
pandas only — no API key needed*

You have been asked to brief your CIO on what other agencies are actually
doing with AI. The OMB inventory is public and sitting on your VM. No AI is
needed for this lab — this is the data literacy underneath it.

## Objectives

By the end of this lab, you will:

- Load, filter, group and summarise a real federal dataset.
- Answer substantive questions about federal AI adoption from evidence.
- Practise the analysis habits every later lab depends on.

## Setup

- **Datasets:** `data/federal_ai_use_cases.csv` (the individually-reported
  portion of OMB's 2025 Federal Agency AI Use Case Inventory — capped
  extract) and `data/federal_ai_cots.csv` (its commercial off-the-shelf
  companion). Provenance in `data/MANIFEST.json`.
- **No API key is needed** — this lab never calls a model.
- **The one rule that never changes:** public or synthetic data only.
- **How this notebook works:** every step is one provided cell — run it with
  Shift+Enter and read what it prints. Cells marked `# YOUR TURN` ask you to
  change a simple value and re-run the cell. Everything runs as shipped, so
  you can never get stuck.

In [1]:
# ▶ Setup — run this cell first (click it, then Shift+Enter).
# It loads the course helper functions used by every step below.
from lab_helpers import *

## Steps

1. **(3 min)** Run the setup cell above, then load the inventory.
2. **(3 min)** Print the shape and columns. How many use cases, how many
   agencies?
3. **(4 min)** Produce the top 10 agencies by use-case count.
4. **(4 min)** Break the inventory down by development stage. What share is
   actually in production?
5. **(4 min)** Which agencies report the most high-impact AI?
6. **(4 min)** Cross-tabulate `topic_area` against `development_stage` and
   read one row aloud.
7. **(4 min)** Which commercial tools appear most often?
8. **(4 min)** Write three findings in the notebook, each with the step that
   produced it.

### Step 1 — Load the inventory (provided)

Run this cell. It loads `data/federal_ai_use_cases.csv` — the
individually-reported portion of OMB's 2025 inventory (capped extract — see
`data/MANIFEST.json` for provenance).

In [2]:
uc = load_ai_inventory()

loaded federal_ai_use_cases.csv: 1,200 rows x 36 columns


### Step 2 — First look: shape, columns, agencies (provided)

Run this cell to print the shape and columns. **How many use cases, and how
many agencies?**

In [3]:
inventory_overview(uc)

shape: (1200, 36) | agencies: 6
columns: ['agency', 'agency_name', 'id', 'use_case_name', 'agency_bureau', 'contact_email', 'is_withheld', 'development_stage', 'is_high_impact', 'HI_justification', 'topic_area', 'classification', 'problem_solved', 'benefits', 'system_outputs', 'operational_date', 'contracting_usage', 'vendor_name', 'have_ato', 'system_name_ato', 'data_description', 'link_to_data', 'has_pii', 'pia_url', 'demographic_features', 'has_custom_code', 'code_url', 'hi_testing_conducted', 'hi_assessment_completed', 'hi_potential_impacts', 'hi_independent_review', 'hi_ongoing_monitoring', 'hi_training_established', 'hi_failsafe_presence', 'hi_appeal_process', 'hi_public_consultation']


### Step 3 — Top 10 agencies by use-case count

Run this cell to rank agencies by how many use cases they reported. Then
change `TOP_N` below and re-run — what changes?

In [4]:
TOP_N = 10   # ← YOUR TURN: how many agencies to show — try 3 or 20, then re-run this cell
top_ai_agencies(uc, top_n=TOP_N)

agency_name
Department of Energy                    340
Department of the Interior              247
Department of Homeland Security         238
Department of Commerce                  223
Department of Justice                   149
Commodity Futures Trading Commission      3


### Step 4 — What share is actually in production? (provided)

Run this cell to break the inventory down by `development_stage`. What share
of use cases is `Deployed`? The cell prints the share two ways — blanks
excluded, and blanks counted in the total. Decide for yourself which you
would quote — and be ready to defend it.

In [5]:
show_deployed_share(uc)

development_stage
Pre-deployment    397
Deployed          347
NaN               223
Pilot             166
Retired            67

Deployed share (of non-blank): 35.5%
Deployed share (of all rows):  28.9%


### Step 5 — Who reports the most high-impact AI? (provided)

Run this cell to keep only the rows flagged `is_high_impact == "High-impact"`
and rank agencies. (Lab 0.1 already showed this column has more than two
values — the cell filters deliberately, it does not just count non-blanks.)

In [6]:
high_impact_agencies(uc)

agency_name
Department of Justice              59
Department of Homeland Security    55
Department of Energy               29

flagged High-impact: 143 of 1200


### Step 6 — Topic area × development stage (provided)

Run this cell to cross-tabulate `topic_area` against `development_stage`,
then read one row aloud to your neighbour: what does it tell you about where
that topic's AI actually is?

In [7]:
topic_stage_crosstab(uc)

development_stage             Deployed  Pilot  Pre-deployment
topic_area                                                   
Admin Functions                     53     16              69
Benefits Processing                  1      0               3
Cybersecurity                        9      1               6
Emergency Mgmt                       5      2               4
Energy and Environment              15     11              40
HR                                   4      1               5
Health and Medical                   1      0               3
IT                                  66     17              39
International Affairs                1      0               1
Law Enforcement                    112      9              52
Other                               24     18              10
Procurement and Finance Mgmt         6      0              10
Science                             34     87             133
Service Delivery                     6      3              11
Transpor

### Step 7 — The commercial tools underneath (provided)

`federal_ai_cots.csv` is the companion file: consolidated commercial
off-the-shelf AI use. Run this cell to see which commercial products appear
most often. Notice the near-duplicates in the product names while you read —
a preview of Lab 6.2.

In [8]:
top_commercial_tools()

Name of Commercial Product or Service Used
Microsoft Copilot                       51
Various                                 24
Microsoft M365 Copilot AI               20
Apple iPhone                            15
M365 Copilot                            10
Google Maps, Apple Maps                  9
Microsoft 365 Copilot GCC                8
Copilot for MS 365                       7
ChatGPT, Claude, Copilot, Perplexity     7
Gemini for Google Workspace              6

rows mentioning a Copilot variant: 211 of 900


### Step 8 — Three findings, each with the step that produced it

Write three findings for your CIO in the markdown cell below. Each must be
one sentence *plus* the step above that produced the number. A finding is
something a decision-maker could act on, not a restated table.

**Your findings (write here):**

1.
2.
3.

## Deliverable

1. The top-10 agency table, the deployed share, the high-impact ranking.
2. The topic × stage cross-tab with one row read aloud.
3. Your three findings, each with the step that produced it.

## Reflection

1. Which single number would most change your CIO's mind?
2. What does this dataset *not* tell you about how well any of it works?
3. Where would a chart have been clearer than a table?

## Debrief (instructor-led)

- Compare deployed shares across the room: who included the 223 blanks in
  the denominator, who excluded them, and who can defend their choice? Both
  are defensible if stated — the blanks themselves are a finding about the
  inventory's completeness.
- Contrast the two rankings on the board: most AI by volume vs most
  *high-impact* AI. They are not the same agencies — which number belongs in
  the CIO briefing?
- Only six agencies appear in this capped extract (the full inventory is
  56). Ask what the cap does to any "top 10" claim before anyone quotes one.

## Troubleshooting

- **A red error mentioning a data file.** The course data pack is incomplete
  — tell your instructor; run the Day-0 healthcheck to confirm.
- **The counts in a step don't add up to the row count.** Blank values are
  left out of most counts by default — Step 4's table shows them as `NaN`.
- **Your numbers differ from your neighbour's.** Check you are both on the
  same shipped extract; this file is capped (see `data/MANIFEST.json`), not
  the full 56-agency inventory.
- **Out-of-order errors after experimenting.** Kernel → Restart & Run All.

---
*Curious about the Python behind these steps? The full code-forward version
of this lab lives in the `For_Python_Programmers/` folder.*